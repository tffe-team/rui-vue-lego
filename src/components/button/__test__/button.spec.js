import {shallowMount, config} from '@vue/test-utils'
import Button from '../Button'

// TODO: Button 组件依赖这个属性，后续考虑优化，因为很多组件都会依赖
config.mocks['$LEGO'] = {
  theme: {
    theme: 'blue',
    lang: 'zh-CN'
  }
}

// 获取包裹器
const wrapper = shallowMount(Button)

test('点击按钮触发 click 事件', () => {
  wrapper.find('button').trigger('click')
  expect(wrapper.emitted().click).toBeTruthy()
})
