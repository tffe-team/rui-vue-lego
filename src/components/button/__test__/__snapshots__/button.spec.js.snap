// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create a button 1`] = `
"<button class=\\"vue-lego-btn vue-lego-btn-[object Object]-success\\">
  <!----> </button>"
`;
